src/Strategy/SwiftPM.hs

```diff
@@ -1,61 +1,129 @@
+  { xCodeProjectFile :: Path Abs File
+  , xCodeProjectDir :: Path Abs Dir
+  , xCodeResolvedFile :: Maybe (Path Abs File)
   }
   deriving (Show, Eq, Ord)
 
-discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has Logger sig m, Has ReadFS rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
 discover dir = context "Swift" $ do
-  projects <- context "Finding projects" $ findProjects dir
-  pure (map mkProject projects)
-
-findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [SwiftPackageProject]
-findProjects = walk' $ \dir _ files -> do
-  let swiftPackageManifestFile = findFileNamed "Package.swift" files
-  let swiftPackageResolvedFile = findFileNamed "Package.resolved" files
-
-  case (swiftPackageManifestFile, swiftPackageResolvedFile) of
-    (Just manifestFile, Just resolvedFile) -> pure ([SwiftPackageProject manifestFile dir (Just resolvedFile)], WalkSkipSome [".build"])
-    (Just manifestFile, Nothing) -> pure ([SwiftPackageProject manifestFile dir Nothing], WalkSkipSome [".build"])
-    -- Package.resolved without Package.swift is not Swift Package Project
-    (Nothing, Just _) -> pure ([], WalkContinue)
-    (Nothing, Nothing) -> pure ([], WalkContinue)
-
-mkProject :: (Has ReadFS sig n, Has Diagnostics sig n) => SwiftPackageProject -> DiscoveredProject n
+  swiftPackageProjects <- context "Finding swift package projects" $ findSwiftPackageProjects dir
+  xCodeProjects <- context "Finding xcode projects using swift package manager" $ findXcodeProjects dir
+  pure $ map mkProject (swiftPackageProjects ++ xCodeProjects)
+
+findSwiftPackageProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [SwiftProject]
+findSwiftPackageProjects = walk' $ \dir _ files -> do
+  let packageManifestFile = findFileNamed "Package.swift" files
+  let packageResolvedFile = findFileNamed "Package.resolved" files
+  case (packageManifestFile, packageResolvedFile) of
+    -- If the Package.swift exists, than it is swift package project.
+    -- Use Package.swift as primary source of truth.
+    (Just manifestFile, resolvedFile) -> pure ([PackageProject $ SwiftPackageProject manifestFile dir resolvedFile], WalkSkipSome [".build"])
+    -- Package.resolved without Package.swift or Xcode project file is not a valid swift project.
+    (Nothing, _) -> pure ([], WalkContinue)
+
+findXcodeProjects :: (Has ReadFS sig m, Has Diagnostics sig m, Has Logger sig m) => Path Abs Dir -> m [SwiftProject]
+findXcodeProjects = walk' $ \dir _ files -> do
+  let xcodeProjectFile = findFileNamed "project.pbxproj" files
+  case xcodeProjectFile of
+    Nothing -> pure ([], WalkContinue)
+    Just projFile -> do
+      resolvedFile <- findFirstResolvedFileRecursively dir
+      isValidXCodeProj <- hasSomeSwiftDeps projFile
```

Comment:
Not all xcode project use swift pm for dependency management. This is to avoid unnecessary noise from analysis.